src/app/pages/traslados/form-traslado/form-traslado.component.ts

```diff
@@ -0,0 +1,284 @@
+        distinctUntilChanged(),
+        map(val => typeof val === 'string' ? val : this.muestraFuncionario(val)),
+        map(nombre => this.filtroFuncionarios(nombre)),
+      );
+  }
+
+  private cambiosDependencia(valueChanges: Observable<any>) {
+    valueChanges.pipe(
+      startWith(''),
+      debounceTime(250),
+      distinctUntilChanged(),
+      switchMap((val) => this.loadDependencias(val)),
+    ).subscribe((response: any) => {
+      this.dependencias = response.queryOptions[0].Id ? response.queryOptions : [];
+      this.getUbicaciones();
+    });
+  }
+
+  private filtroFuncionarios(nombre: string): TerceroCriterioContratista[] {
+    if (nombre.length >= 4 && Array.isArray(this.funcionarios)) {
+      const valorFiltrado = nombre.toLowerCase();
+      return this.funcionarios.filter(contr => this.muestraFuncionario(contr).toLowerCase().includes(valorFiltrado));
+    } else return [];
+  }
+
+  private loadDependencias(text: string) {
+    const queryOptions$ = text.length > 3 ?
+      this.oikosHelper.getDependencias(text) :
+      new Observable((obs) => { obs.next([{}]); });
+    return combineLatest([queryOptions$]).pipe(
+      map(([queryOptions_$]) => ({
+        queryOptions: queryOptions_$,
+      })),
+    );
+  }
+
+  private loadFuncionarios(): Promise<void> {
+    return new Promise<void>(resolve => {
+      this.tercerosHelper.getTercerosByCriterio('funcionarios').toPromise().then(res => {
+        this.funcionarios = res;
+        resolve();
+      });
+    });
+  }
+
+  private validarTercero(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const valor = control.value;
+      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '' ? true : false;
+      const checkInvalidString = typeof (valor) === 'string' && valor !== '' ? true : false;
+      const checkInvalidTercero = typeof (valor) === 'object' && !valor.Tercero ? true : false;
+      return checkStringLength ? { errorLongitudMinima: true } :
+        checkInvalidString || checkInvalidTercero ? { terceroNoValido: true } : null;
+    };
+  }
+
+  private checkValidness: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
+    const errors = control.get('origen.tercero').value !== '' &&
+      control.get('origen.tercero').value === control.get('destino.tercero').value;
+    errors && this.formTraslado ? control.get('destino.tercero').setErrors({ 'mismoTercero': true }) : null;
+    return null;
+  }
+
+  private validateObjectCompleter(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const valor = control.value;
+      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '' ? true : false;
+      const checkInvalidString = typeof (valor) === 'string' && valor !== '' ? true : false;
+      const checkInvalidObject = typeof (valor) === 'object' && !valor.Id ? true : false;
```

Comment:
Uso innecesario de ternas
```suggestion
      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '';
      const checkInvalidString = typeof (valor) === 'string' && valor !== '';
      const checkInvalidObject = typeof (valor) === 'object' && !valor.Id;
```